blogs/2022/03/28/increase-productivity-with-containers.md

```diff
@@ -0,0 +1,89 @@
+
+This brings us to the concept of a development container: a container that runs a full-featured development environment. A dev container houses its own app and dependencies, such as the required tools, libraries, and runtimes. In the diagram below, you can see that these dependencies exist in the containers and not on the host machine, which means that you can seamlessly switch between tech stacks in no time at all.
+
+![A diagram showing containers with different tech stacks](container-diagram.png)
+Image Source: Development Containers in Education with Visual Studio Code
+
+To provide a way to create and connect to dev containers, VS Code unveiled the [Remote -  Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) in 2019. This extension provides the illusion of local development by leveraging the full power of dev containers, all while never leaving the comfort of VS Code. With over 11 million installs of the extension, it got us thinking: what if you could have a dev container that is hosted in the cloud?
+
+## Moving to the cloud
+
+Let’s be real: everything is moving to the cloud, so why should your development environment be any different? Well, for one, the cloud is vulnerable to attacks. For two, it seems like we’re constantly hearing about major outages. For three, it can be pricey. For four… wait, isn’t the cloud supposed to be a good thing?
+
+With all these reasons to be weary of the cloud, it can be uncomfortable to think about something as important as our development environment being hosted there.  So let’s stay comfortable! Let’s continue relying on our 6-year-old laptop that makes really weird whirring noises if you try to open an email too quickly, let alone try to build a project. And we’ll just comfortably wait for the inevitability of when the laptop crashes, and then once we get our new laptop, we have to rebuild our dev environment and try to remember exactly how we had it configured.
+
+![A meme showing a dog saying "This is fine" while sitting in a room on fire](this-is-fine-meme.png)
+
+Turns out staying in our comfort zone doesn’t sound so peachy either.
+
+While there are inherent risks with the cloud, it also comes with major benefits: scalability, faster performance, and easier maintenance, just to name a few. Hosting in the cloud gives you access to another machine’s resources that can then be spun up quickly and available anywhere. Combine that with the benefits of container-based development, and developers can be up and coding in the blink of an eye.
```

Comment:
Rather than implying we just have to accept those risks from cloud-based services in the first paragraph, could we also mention how Codespaces tries to avoid them, further increasing users' confidence?

Reply:
great point! i added a couple paragraphs about this in the next section when i talk about codespaces - let me know if you think it would benefit to add in this section too though!

Reply:
Would you be able to point to which paragraphs address this? Want to make sure I'm not missing it!

Reply:
tagged you in the section :)